pkg/apis/resource/v1alpha1/pipelineresource_validation_test.go

```diff
@@ -32,86 +32,26 @@ func TestResourceValidation_Invalid(t *testing.T) {
 		name string
 		res  *v1alpha1.PipelineResource
 		want *apis.FieldError
-	}{
-		{
-			name: "storage with no type",
-			res: &v1alpha1.PipelineResource{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "temp",
-				},
-				Spec: v1alpha1.PipelineResourceSpec{
-					Type: v1alpha1.PipelineResourceTypeStorage,
-					Params: []v1alpha1.ResourceParam{{
-						Name: "no-type-param", Value: "something",
-					}},
-				},
+	}{{
+		name: "invalid resource type",
+		res: &v1alpha1.PipelineResource{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "temp",
 			},
-			want: apis.ErrMissingField("spec.params.type"),
-		}, {
-			name: "storage with unimplemented type",
-			res: &v1alpha1.PipelineResource{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "temp",
-				},
-				Spec: v1alpha1.PipelineResourceSpec{
-					Type: v1alpha1.PipelineResourceTypeStorage,
-					Params: []v1alpha1.ResourceParam{{
-						Name: "type", Value: "not-implemented-yet",
-					}},
-				},
+			Spec: v1alpha1.PipelineResourceSpec{
+				Type: "not-supported",
 			},
-			want: apis.ErrInvalidValue("not-implemented-yet", "spec.params.type"),
-		}, {
-			name: "storage with gcs type with no location param",
-			res: &v1alpha1.PipelineResource{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "temp",
-				},
-				Spec: v1alpha1.PipelineResourceSpec{
-					Type: v1alpha1.PipelineResourceTypeStorage,
-					Params: []v1alpha1.ResourceParam{{
-						Name: "type", Value: "gcs",
-					}},
-				},
-			},
-			want: apis.ErrMissingField("spec.params.location"),
-		}, {
-			name: "storage with gcs type with empty location param",
-			res: &v1alpha1.PipelineResource{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "temp",
-				},
-				Spec: v1alpha1.PipelineResourceSpec{
-					Type: v1alpha1.PipelineResourceTypeStorage,
-					Params: []v1alpha1.ResourceParam{{
-						Name: "type", Value: "gcs",
-					}, {
-						Name: "location", Value: "",
-					}},
-				},
-			},
-			want: apis.ErrMissingField("spec.params.location"),
-		}, {
-			name: "invalid resource type",
-			res: &v1alpha1.PipelineResource{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "temp",
-				},
-				Spec: v1alpha1.PipelineResourceSpec{
-					Type: "not-supported",
-				},
-			},
-			want: apis.ErrInvalidValue("spec.type", "not-supported"),
-		}, {
-			name: "missing spec",
-			res: &v1alpha1.PipelineResource{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "temp",
-				},
```

Comment:
let's keep these test cases (missing spec and invalid type) around

Reply:
sorry i think the (missing spec and invalid type) two are retained, these changes might have made it confusing after rebasing and merging multiple PRs from branches:/